### ISSUE TITLE
Does sfacctd support aggregation of Q-in-Q traffic ?

### ISSUE THREAD

**Issue by kanix29** [pmacct/pmacct#626]
Hi Paolo (@paololucente),
We would like to collect traffic with multiple vlan tags(Q-in-Q) using sfacctd, but such traffic does not appear in output csv files.

So we would like to clarify whether sfacctd supports aggregation of such Q-in-Q traffic. If yes, do you have any idea of why we could not see traffic data of them on the output csv file? 

A config is below:

```
daemonize: true
pidfile: /var/run/sfacctd.pid
syslog: daemon
aggregate: src_mac, dst_mac, vlan, cos, etype, src_as, dst_as, peer_src_ip, peer_dst_ip, in_iface, out_iface, src_host, src_net, dst_host, dst_net, src_mask, dst_mask, src_port, dst_port, tcpflags, proto, tos, sampling_rate, timestamp_start, timestamp_end, timestamp_arrival
sfacctd_port: 50001

plugins: print
print_refresh_time: 300
print_history: 5m
print_output: csv
print_output_file: /var/csvs/usws.sflow.%Y%m%d-%H%M.csv
print_history_roundoff: m

print_cache_entries: 2000000
print_output_file_append: true

plugin_pipe_size:  10240000
plugin_buffer_size: 10240
```

Version:

```
 $ sfacctd -V
sFlow Accounting Daemon, sfacctd 1.7.8-git [20220712-0 (327183a)]

Arguments:
 '--enable-l2' '--enable-traffic-bins' '--enable-bgp-bins' '--enable-bmp-bins' '--enable-st-bins'

Libs:
cdada 0.4.0
libpcap version 1.4.0

Plugins:
memory
print
nfprobe
sfprobe
tee

System:
Linux 2.6.32-431.17.1.el6.x86_64 #1 SMP Fri Apr 11 17:27:00 EDT 2014 x86_64

Compiler:
gcc 4.4.7

For suggestions, critics, bugs, contact me: Paolo Lucente <paolo@pmacct.net>.
```

Thanks in advance.

**Comment by paololucente** [comment]
Hi @kanix29 ,

> So we would like to clarify whether sfacctd supports aggregation of such Q-in-Q traffic. If yes, do you have any idea of why we could not see traffic data of them on the output csv file?

I'd need one clarification: what are you looking for / what is not working? 1) A missing primitive for the second VLAN tag? Or is it a problem with the payload that is not decoded well? Anyway, both issues are largely solvable; could you send me by unicast email a sample of your sFlow q-in-q flows? It will help me with the fixing / dev'ing. 

Paolo

**Comment by kanix29** [comment]
Hi, @paololucente !

> I'd need one clarification: what are you looking for / what is not working?
The problem is that the payload is not decoded well. Therefore, certain interface does not appear in output csv files.

I sent you an email with a sample of sFlow q-in-q flows, so please check it.

**Comment by paololucente** [comment]
Conversation continued with @kanix29 via unicast email. This was resolved by commits a2e27f7af0a8ae7c3ea970b368961a384e6cf457 , d8bfd60e71b2d1bd695c1a8c72cd6f2ea6d255af and 57e5b892b8d16e964fb7abebe211f74c85d66cd4 .
